morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -1,81 +1,159 @@
+import Ouroboros.Network.Protocol.Handshake.Type (Handshake)
+import qualified Ouroboros.Network.Protocol.Handshake.Type as Handshake
+import qualified Ouroboros.Network.Protocol.Trans.Hello.Type as Hello
+import qualified Ouroboros.Network.Protocol.TxSubmission.Type as TxSubmission
+import Ouroboros.Network.Snocket (FileDescriptor, LocalAddress (LocalAddress))
+import Ouroboros.Network.Subscription (ConnectResult (ConnectSuccess, ConnectSuccessLast, ConnectValencyExceeded))
+import Ouroboros.Network.TxSubmission.Inbound (ProcessedTxCount (ProcessedTxCount), TraceTxSubmissionInbound (TraceTxInboundCanRequestMoreTxs, TraceTxInboundCannotRequestMoreTxs, TraceTxInboundTerminated, TraceTxSubmissionCollected, TraceTxSubmissionProcessed))
+import Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbound (TraceControlMessage, TraceTxSubmissionOutboundRecvMsgRequestTxs, TraceTxSubmissionOutboundSendMsgReplyTxs))
+import Prettyprinter (Doc, Pretty (..), comma, concatWith, hsep, punctuate, viaShow, (<+>))
 
 class MPretty a where
   mpretty :: a -> Doc ann
 
-instance (HashAlgorithm h, BftCrypto c) => MPretty (MorphoBlock h c) where
-  mpretty MorphoBlock {..} =
-    "("
-      <> mpretty morphoPrev
-      <> "->"
-      <> mpretty morphoHeaderHash
-      <> ","
-      <> mpretty morphoSlotNo
-      <> ","
-      <> mpretty morphoTxs
+instance MPretty (HeaderHash blk) => MPretty (MorphoState blk) where
+  mpretty (MorphoState lastCkpt ckptAt votes tip) =
+    "MorphoState(morphoTip:"
+      <+> mpretty tip
+      <> ", lastCheckpoint:"
+      <+> mpretty lastCkpt
+      <> ", checkpointAt:"
+      <+> mpretty ckptAt
+      <> ", currentVotes:"
+      <+> mpretty votes
       <> ")"
-    where
-      MorphoHeader {..} = morphoHeader
-      MorphoStdHeader {..} = morphoHeaderStd
-      MorphoBody {..} = morphoBody
 
-instance
-  (HashAlgorithm h, BftCrypto c) =>
-  MPretty (Header (MorphoBlock h c))
-  where
+instance MPretty PublicKey where
+  mpretty (PublicKey bytes) =
+    "PublicKey(" <> mpretty bytes <> ")"
+
+instance MPretty Bytes where
+  mpretty = pretty . Text.take 8 . bytesToHex
+
+instance MPretty (MorphoBlock h c) where
+  mpretty
+    MorphoBlock
+      { morphoHeader = MorphoHeader {morphoHeaderHash, morphoHeaderStd},
+        morphoBody
+      } =
+      "MorphoBlock(hash:"
+        <+> mpretty morphoHeaderHash
+        <> ", prevHash:"
+        <+> mpretty (morphoPrev morphoHeaderStd)
+        <> ", slot:"
+        <+> mpretty (morphoSlotNo morphoHeaderStd)
+        <> ", blockNo:"
+        <+> mpretty (morphoBlockNo morphoHeaderStd)
+        <> ", body:"
+        <+> mpretty morphoBody
+        <> ")"
+
+instance MPretty MorphoBody where
+  mpretty (MorphoBody txs) =
+    "MorphoBody("
+      <> hsep (punctuate comma (map mpretty txs))
+      <> ")"
+
+instance MPretty (Header (MorphoBlock h c)) where
   mpretty MorphoHeader {..} =
     "("
       <> mpretty morphoPrev
```

Comment:
This instance should be improved, an example of how it looks is

> (BlockHash(3e867db0)->c337db48,13948056)

which is very ambiguous. Let's use the same format as the others instead.